src/sylveon/_classes/_upload.py

```diff
@@ -0,0 +1,74 @@
+"""Functions for uploading assignments to Gradescope."""
+
+import requests
+from bs4 import BeautifulSoup
+from requests_toolbelt.multipart.encoder import MultipartEncoder
+import mimetypes
+import io
+import pathlib
+
+
+def upload_assignment(
+    session: requests.Session,
+    course_id: str,
+    assignment_id: str,
+    *files: io.TextIOWrapper,
+    leaderboard_name: str | None = None,
+) -> str | None:
+    """Uploads given file objects to the specified assignment on Gradescope.
+
+    Args:
+        session (requests.Session): The session object to use for making HTTP requests.
+        course_id (str): The ID of the course on Gradescope.
+        assignment_id (str): The ID of the assignment on Gradescope.
+        *files (io.TextIOWrapper): Variable number of file objects to upload.
+        leaderboard_name (str | None, optional): The name of the leaderboard. Defaults to None.
```

Comment:
just curious, what exactly is the leaderboard?

Reply:
Leaderboards can be set up for programming assignments based on the autograder for programming assignments: https://gradescope-autograders.readthedocs.io/en/latest/leaderboards/

To be honest, I've never been in a course that used the feature. But for the test assignment I enabled it to play around with it.